src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -462,13 +463,11 @@ private BoundExpression ConvertSwitchExpression(BoundUnconvertedSwitchExpression
             for (int i = 0, n = source.SwitchArms.Length; i < n; i++)
             {
                 var oldCase = source.SwitchArms[i];
-                Debug.Assert(oldCase.Syntax is SwitchExpressionArmSyntax);
-                var binder = GetRequiredBinder(oldCase.Syntax);
```

Comment:
>var binder = GetRequiredBinder(oldCase.Syntax); [](http://example.com/codeflow?start=16&length=47)

I guess reverting this change feels fine, but we probably need to open an issue to follow up on other target-typed expressions to make sure they are also using the "right" binder for converting to the target type.

Reply:
Afaict this is the only place that needed this adjustment as mentioned in the linked PR. Now that we're using the original binder we don't need to do any of that.

Reply:
> Now that we're using the original binder we don't need to do any of that.

This only covers target-typed new. I am talking about "an issue to follow up on **other** target-typed expressions"

Reply:
https://github.com/dotnet/roslyn/issues/61179